### PR TITLE
Fix tensor saving and loading twice.

### PR DESCRIPTION
**Pull request by vladov3000**
Solves #6294. See [this comment](https://github.com/tinygrad/tinygrad/issues/6294#issuecomment-2382370590) for the details. TLDR; disk devices keep around unlinked files and don't create new files.

This is my first contribution, so there may be a few rough edges. Namely, mypy has poor data flow analysis for `Optional`, so I used sentinel values instead. And no, it doesn't make sense for `self.fd` to be 0 since `mmap`ing stdin will give an error. Also, don't know if multiple assignment is ok here.

I reproduced the given script and got it working. I also added a unit test for this edge case.

Only 1 new line of code + 3 lines of test code.

### CODE REVIEW

File: tinygrad/runtime/ops_disk.py [hunk 1]
```diff
@@ -71,21 +71,22 @@ class DiskDevice(Compiled):
   def __init__(self, device:str):
     if not DiskDevice._tried_io_uring_init: self._iouring_setup()
 
-    self.size: Optional[int] = None
-    self.count = 0
+    self.size = self.fd = self.count = 0
```
Comment (geohot) [inline tinygrad/runtime/ops_disk.py:74]: Why did you remove a type? Don't save lines like this!!

Comment (vladov3000) [reply tinygrad/runtime/ops_disk.py:74]: I removed the type because I don’t think annotating the type for an int assignment is necessary. The line above, ‘self.count = 0’ does not have a type annotation either. 

For more context, I changed the type of self.size from Optional[int] to an int where size = 0 instead of size = None represents that the file has not been memory mapped yet. This is because extra asserts/casts/ifs are required to work around mypy and there is a lack of helper functions like map or apply from Haskell. So I used an int and cut down on the code size. If you aren’t ok with this, let me know and I will use Nones instead of 0s in future PRs.
